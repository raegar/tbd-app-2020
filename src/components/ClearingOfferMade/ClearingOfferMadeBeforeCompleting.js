import React, {useState} from "react";
import TextBox from "../TextBox/TextBox";
import "./ClearingOfferMade.css";
import { Container, Col, Row } from "react-bootstrap";
import UCPButton from "../Buttons/UCPButton";


function ClearingOfferMadeBeforeCompleting() {
	const [email, setEmail] = useState(global.ApplicationFormData.email);
	const [course, setCourse] = useState(global.ApplicationFormData.course);

	function downloadFile() {
    	let fileName = "studentDetails.json"; 

    	let saveData = (function () { 
    		let a = document.createElement("a"); 
    		document.body.appendChild(a); 
    		a.style = "display: none"; 
    		return function (fileName) { 
    			var json = JSON.stringify(global.ApplicationFormData), 
    				blob = new Blob([json], {type: "octet/stream"}), 
    				url = window.URL.createObjectURL(blob); 
    			a.href = url; 
    			a.download = fileName; 
    			a.click(); 
    			window.URL.revokeObjectURL(url); 
    		}; 
    	}()); 
    	saveData(fileName);
	}

	return ( 
		<Container>
			<h1>Before completing the offer....</h1>
			<p>Please state the following</p>
			<h3>40. Please confirm email offer will be send to</h3>
			<TextBox id="emailConfirm" placeholder={global.ApplicationFormData.email} value={email}/>
			<h3>41. Please enter course applied for</h3>
			<TextBox id="courseApplied" value={course}/>
			<Row>
				<Col>
					<br></br>
					<UCPButton to='/ClearingOfferMadeSecondPage'
						primary="True"
						className="mediumbutton"
						buttonText="Go Back"
						href="/ClearingOfferMadeSecondPage"
					/>
				</Col>

				<Col>
					<br></br>
					<div onClick={downloadFile}>
						<UCPButton to ='/'
							primary="True"
							className="mediumbutton"
							buttonText="Confirm and Continue"
							href="/"
						/>
					</div>
				</Col>
			</Row>
		</Container>
	);
}

export default ClearingOfferMadeBeforeCompleting;