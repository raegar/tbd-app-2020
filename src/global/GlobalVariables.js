global.ApplicationFormData = {
	levelThreeQualifications: {
		qual: [],
		subject: [],
		grade: [],
		date: [],
	}
};


global.userType = "student";
global.staffFirstName = "";
global.staffLastName = "";
global.PageThreeSkip = false;

