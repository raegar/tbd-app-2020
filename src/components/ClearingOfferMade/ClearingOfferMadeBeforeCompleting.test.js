import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import ClearingOfferMadeBeforeCompleting from './ClearingOfferMadeBeforeCompleting';
import { BrowserRouter as Router} from 'react-router-dom';
import '../../global/GlobalVariables.js';

//cleans up 
afterEach(cleanup);

it("Renders ClearingOfferMadeBeforeCompleting", () => {
    global.ApplicationFormData.email = "Test@email.com";
    const { asFragment } = render(<Router><ClearingOfferMadeBeforeCompleting /></Router>);
    expect(asFragment()).toMatchSnapshot();
});