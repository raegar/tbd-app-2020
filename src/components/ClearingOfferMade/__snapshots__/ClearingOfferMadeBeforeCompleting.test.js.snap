// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ClearingOfferMadeBeforeCompleting 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      Before completing the offer....
    </h1>
    <p>
      Please state the following
    </p>
    <h3>
      40. Please confirm email offer will be send to
    </h3>
    <div
      class="mobiletextbox"
    >
      <label
        for="emailConfirm"
      />
      <input
        id="emailConfirm"
        maxlength="30"
        minlength="0"
        placeholder="Test@email.com"
        size="20"
        type="text"
      />
    </div>
    <h3>
      41. Please enter course applied for
    </h3>
    <div
      class="mobiletextbox"
    >
      <label
        for="courseApplied"
      />
      <input
        id="courseApplied"
        maxlength="30"
        minlength="0"
        placeholder=""
        size="20"
        type="text"
      />
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <br />
        <a
          href="/ClearingOfferMadeSecondPage"
        >
          <button
            class="mediumbutton"
            data-testid="ucpButton"
            href="/ClearingOfferMadeSecondPage"
            primary="True"
            type="button"
          >
            Go Back​
          </button>
        </a>
      </div>
      <div
        class="col"
      >
        <br />
        <div>
          <a
            href="/"
          >
            <button
              class="mediumbutton"
              data-testid="ucpButton"
              href="/"
              primary="True"
              type="button"
            >
              Confirm and Continue​
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Renders ClearingOfferMadeFifthPage 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <h1>
      Before completing the offer....
    </h1>
    <p>
      Please state the following
    </p>
    <h3>
      40. Please confirm email offer will be send to
    </h3>
    <div
      class="mobiletextbox"
    >
      <label
        for="emailConfirm"
      />
      <input
        id="emailConfirm"
        maxlength="30"
        minlength="0"
        placeholder="Test@email.com"
        size="20"
        type="text"
      />
    </div>
    <h3>
      41. Please enter course applied for
    </h3>
    <div
      class="mobiletextbox"
    >
      <label
        for="courseApplied"
      />
      <input
        id="courseApplied"
        maxlength="30"
        minlength="0"
        placeholder=""
        size="20"
        type="text"
      />
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <br />
        <a
          href="/ClearingOfferMadeSecondPage"
        >
          <button
            class="mediumbutton"
            data-testid="ucpButton"
            href="/ClearingOfferMadeSecondPage"
            primary="True"
            type="button"
          >
            Go Back​
          </button>
        </a>
      </div>
      <div
        class="col"
      >
        <br />
        <div>
          <a
            href="/"
          >
            <button
              class="mediumbutton"
              data-testid="ucpButton"
              href="/"
              primary="True"
              type="button"
            >
              Confirm and Continue​
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
